Shield.Communication/Services/Wifi.cs

```diff
@@ -0,0 +1,297 @@
+                {
+                    var b = reader.ReadByte();
+                    sb.Append((char) b);
+                }
+
+                var msg = sb.ToString();
+
+                if (msg.StartsWith("VS:"))
+                {
+                    msg = msg.Substring(3);
+                    var colon = msg.IndexOf(':');
+                    var typename = msg.Substring(0, colon);
+                    var equals = msg.IndexOf('=');
+                    var ip = msg.Substring(colon + 1, equals - colon - 1);
+                    var name = msg.Substring(equals + 1);
+                    var colon2 = name.IndexOf(':');
+
+                    if (colon2 > -1)
+                    {
+                        name = name.Substring(0, colon2);
+                    }
+
+                    if (string.IsNullOrWhiteSpace(name))
+                    {
+                        name = $"({typename}):{ip}";
+                    }
+
+                    var port = peer.Port;
+                    var portColon = ip.IndexOf(':');
+                    if (portColon > -1)
+                    {
+                        port = ip.Substring(portColon + 1);
+                        ip = ip.Substring(0, portColon);
+                    }
+
+                    peer.IP = ip;
+                    peer.Name = name;
+                    peer.Message = msg;
+                    peer.Pinged = DateTime.Now;
+                    peer.OriginalPort = peer.Port;
+                    peer.Port = port;
+
+                    var connection = new Connection(name, peer);
+                    if (!clients.ContainsKey(ip))
+                    {
+                        clients[ip] = connection;
+                    }
+                }
+            }
+            catch (Exception exception)
+            {
+                // If this is an unknown status it means that the error is fatal and retry will likely fail.
+                if (SocketError.GetStatus(exception.HResult) == SocketErrorStatus.Unknown)
+                {
+                    throw;
+                }
+            }
+        }
+
+        private async Task<RemotePeer> GetHostNameAndService(object source)
+        {
+            var result = source as RemotePeer;
+
+            if (result != null)
+            {
+                return result;
+            }
+
+            var peer = source as PeerInformation;
+            if (peer != null)
+            {
+                return new RemotePeer(null, peer.HostName, "1");
+            }
+
+            var deviceInfo = source as DeviceInformation;
+            if (deviceInfo != null)
+            {
+                var service = await RfcommDeviceService.FromIdAsync(deviceInfo.Id);
+                return new RemotePeer(null, service.ConnectionHostName, service.ConnectionServiceName);
+            }
+
+            var ip = source as string;
+            if (ip != null)
+            {
+                var iponly = ip;
+                var port = "1235";
```

Comment:
This seems to match the const value. If so, there should only be one source of truth.